packages/cosmic-swingset/x/swingset/ibc.go

```diff
@@ -112,99 +112,305 @@ func (ch channelHandler) Receive(ctx *ControllerContext, str string) (ret string
 ) error {
-	_, err := am.keeper.CallToController(`{"type":"FIXME_IBC_CHAN_OPEN_ACK"}`)
+
+	event := channelOpenAckEvent{
+		Type:                "IBC",
+		Event:               "ackOpen",
+		PortID:              portID,
+		ChannelID:           channelID,
+		CounterpartyVersion: counterpartyVersion,
+	}
+
+	bytes, err := json.Marshal(&event)
+	if err != nil {
+		return err
+	}
+
+	_, err = am.keeper.CallToController(string(bytes))
 	return err
 }
 
+type channelOpenConfirmEvent struct {
+	Type      string `json:"type"`  // IBC
+	Event     string `json:"event"` // openConfirm
+	PortID    string `json:"portID"`
+	ChannelID string `json:"channelID"`
+}
+
 func (am AppModule) OnChanOpenConfirm(
 	ctx sdk.Context,
 	portID,
 	channelID string,
 ) error {
-	_, err := am.keeper.CallToController(`{"type":"FIXME_IBC_CHAN_OPEN_CONFIRM"}`)
+	event := channelOpenAckEvent{
+		Type:      "IBC",
+		Event:     "openConfirm",
+		PortID:    portID,
+		ChannelID: channelID,
+	}
+
+	bytes, err := json.Marshal(&event)
+	if err != nil {
+		return err
+	}
+
+	_, err = am.keeper.CallToController(string(bytes))
 	return err
 }
 
+type channelCloseInitEvent struct {
+	Type      string `json:"type"`  // IBC
+	Event     string `json:"event"` // channelCloseInit
+	PortID    string `json:"portID"`
+	ChannelID string `json:"channelID"`
+}
+
 func (am AppModule) OnChanCloseInit(
 	ctx sdk.Context,
 	portID,
 	channelID string,
 ) error {
-	_, err := am.keeper.CallToController(`{"type":"FIXME_IBC_CHAN_CLOSE_INIT"}`)
+	event := channelCloseInitEvent{
+		Type:      "IBC",
+		Event:     "channelCloseInit",
+		PortID:    portID,
+		ChannelID: channelID,
+	}
+
+	bytes, err := json.Marshal(&event)
+	if err != nil {
+		return err
+	}
+
+	_, err = am.keeper.CallToController(string(bytes))
 	return err
 }
 
+type channelCloseConfirmEvent struct {
+	Type      string `json:"type"`  // IBC
+	Event     string `json:"event"` // channelCloseConfirm
+	PortID    string `json:"portID"`
+	ChannelID string `json:"channelID"`
+}
+
 func (am AppModule) OnChanCloseConfirm(
 	ctx sdk.Context,
 	portID,
 	channelID string,
 ) error {
-	_, err := am.keeper.CallToController(`{"type":"FIXME_IBC_CHAN_CLOSE_CONFIRM"}`)
+	event := channelCloseConfirmEvent{
+		Type:      "IBC",
+		Event:     "channelCloseConfirm",
+		PortID:    portID,
+		ChannelID: channelID,
+	}
+
+	bytes, err := json.Marshal(&event)
+	if err != nil {
+		return err
+	}
+
+	_, err = am.keeper.CallToController(string(bytes))
 	return err
 }
 
+type receivePacketEvent struct {
+	Type   string              `json:"type"`  // IBC
+	Event  string              `json:"event"` // receivePacket
+	Packet channeltypes.Packet `json:"packet"`
+}
+
 func (am AppModule) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 ) (*sdk.Result, error) {
-	_, err := am.keeper.CallToController(`{"type":"FIXME_IBC_RECV_PACKET"}`)
-	return nil, err
+	event := receivePacketEvent{
+		Type:   "IBC",
+		Event:  "receivePacket",
+		Packet: packet,
```

Comment:
We may need to marshal the `packet.GetData()` as a base64 string, too.  Not sure how to best describe that.